src/NServiceBus.Core/Routing/DistributionPolicy.cs

```diff
@@ -13,10 +13,11 @@ public void SetDistributionStrategy(string endpointName, DistributionStrategy di
         public DistributionStrategy GetDistributionStrategy(string endpointName)
         {
             DistributionStrategy configuredStrategy;
-            return configuredStrategies.TryGetValue(endpointName, out configuredStrategy) ? configuredStrategy : defaultStrategy;
+            return configuredStrategies.TryGetValue(endpointName, out configuredStrategy) 
```

Comment:
@Scooletz a lambda that doesn't include a closure will be cached so that it only created once and reused.

While I didn't look at the resulting IL, I did run them through a benchmark using BenchmarkDotNet The results I posted below, look at the `ConcurrentDictionaryKeyExistsFunc` line. It shows no GC and no allocated bytes.